humility-dump-agent/src/udp.rs

```diff
@@ -79,6 +79,37 @@ impl<'a> UdpDumpAgent<'a> {
 
         Ok(reply)
     }
+
+    pub fn check_imageid(&mut self, image_id: &Vec<u8>) -> Result<()> {
```

Comment:
It looks like we always call `UdpDumpAgent::new` followed by `dump_agent.check_imageid`, so maybe just fold this into the constructor (making it return a `Result<Self>`)?